sechub-wrapper-prepare/src/main/java/com/mercedesbenz/sechub/wrapper/prepare/modules/WrapperSkopeo.java

```diff
@@ -0,0 +1,150 @@
+package com.mercedesbenz.sechub.wrapper.prepare.modules;
+
+import static com.mercedesbenz.sechub.wrapper.prepare.cli.PrepareWrapperEnvironmentVariables.*;
+import static com.mercedesbenz.sechub.wrapper.prepare.cli.PrepareWrapperKeyConstants.KEY_PDS_PREPARE_AUTHENTICATION_FILE_SKOPEO;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.mercedesbenz.sechub.commons.pds.ProcessAdapter;
+
+@Component
+public class WrapperSkopeo extends WrapperTool {
+
+    @Value("${" + KEY_PDS_PREPARE_AUTHENTICATION_FILE_SKOPEO + ":authentication.json}")
+    String pdsPrepareAuthenticationFileSkopeo;
+
+    void download(SkopeoContext context) throws IOException {
+        if (!context.getCredentialMap().isEmpty()) {
+            login(context);
+        }
+
+        ProcessBuilder builder = buildProcessDownload(context);
+        ProcessAdapter process = null;
+
+        try {
+            process = processAdapterFactory.startProcess(builder);
+        } catch (IOException e) {
+            throw new IOException("Error while download with Skopeo from: " + context.getLocation(), e);
+        }
+
+        waitForProcessToFinish(process);
+    }
+
+    @Override
+    void cleanUploadDirectory(String uploadDirectory) throws IOException {
+        ProcessBuilder builder = buildProcessClean(uploadDirectory);
+        ProcessAdapter process = null;
+
+        try {
+            process = processAdapterFactory.startProcess(builder);
+        } catch (IOException e) {
+            throw new IOException("Error while cleaning authentication file.", e);
+        }
+
+        waitForProcessToFinish(process);
+    }
+
+    private void login(SkopeoContext context) throws IOException {
+        ProcessBuilder builder = buildProcessLogin(context);
+        exportEnvironmentVariables(builder, context.getCredentialMap());
+        ProcessAdapter process = null;
+
+        try {
+            process = processAdapterFactory.startProcess(builder);
+        } catch (IOException e) {
+            throw new IOException("Error while login with Skopeo to: " + context.getLocation(), e);
+        }
+
+        waitForProcessToFinish(process);
+    }
+
+    private ProcessBuilder buildProcessLogin(SkopeoContext context) {
+        List<String> commands = new ArrayList<>();
+
+        String location = transformLocationForLogin(context.getLocation());
+        File uploadDir = Paths.get(context.getUploadDirectory()).toAbsolutePath().toFile();
+
+        // /bin/bash -c is needed to interpret the $USERNAME and $PASSWORD as
+        // environment variables
+        commands.add("/bin/bash");
+        commands.add("-c");
+        commands.add("skopeo login " + location + " --username $" + PDS_PREPARE_CREDENTIAL_USERNAME + " --password $" + PDS_PREPARE_CREDENTIAL_PASSWORD
```

Comment:
I would recommend to add each command and parameter individually like in the `buildProcessDownload` method below. Maybe even remove the command "/bin/bash -c". This prevents input from being interpreted by the shell. Is there another way than passing the username and password via the command line, to prevent them from appearing in the process name?